internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -332,6 +332,11 @@ func (r *Reconciler) computeDesiredMachineSet(ctx context.Context, deployment *c
 	desiredMS.Spec.Template.Spec.NodeDrainTimeout = deployment.Spec.Template.Spec.NodeDrainTimeout
 	desiredMS.Spec.Template.Spec.NodeDeletionTimeout = deployment.Spec.Template.Spec.NodeDeletionTimeout
 	desiredMS.Spec.Template.Spec.NodeVolumeDetachTimeout = deployment.Spec.Template.Spec.NodeVolumeDetachTimeout
+	if deployment.Spec.MachineNamingStrategy != nil && deployment.Spec.MachineNamingStrategy.Template != "" {
```

Comment:
I think the field in desiredMS should be set to nil if it is nil in the MD, otherwise removing a configured strategy from a MD wouldn't have any effect.

Is there a reason why we can't simply do this?
```go
desiredMS.Spec.MachineNamingStrategy = deployment.Spec.MachineNamingStrategy
```